volatility3/framework/symbols/linux/extensions/hash.py

```diff
@@ -0,0 +1,100 @@
+# This file is Copyright 2022 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+from typing import Dict, Tuple
+
+from volatility3.framework import exceptions, constants
+from volatility3.framework import objects, interfaces
+from volatility3.framework.objects import utility
+
+
+class bash_hash_table(objects.StructType):
+    def __init__(
+        self,
+        context: interfaces.context.ContextInterface,
+        type_name: str,
+        object_info: interfaces.objects.ObjectInformation,
+        size: int,
+        members: Dict[str, Tuple[int, interfaces.objects.Template]],
+    ) -> None:
+
+        super().__init__(
+            context=context,
+            type_name=type_name,
+            object_info=object_info,
+            size=size,
+            members=members,
+        )
+        self.context = context
+        self.layer_name = self.vol.layer_name
+        self.symbol_table_name = self.get_symbol_table_name()
+
+    def is_valid(self):
+        try:
+            bucket_array = self.bucket_array
+            nbuckets = self.nbuckets
+            nentries = self.nentries
+        except exceptions.InvalidAddressException:
+            return False
+
+        if not bucket_array:
+            return False
+
+        if not nbuckets or nbuckets != 64:
+            return False
+
+        if not nentries or nentries <= 1:
+            return False
+
+        return True
+
+    def __iter__(self):
+        if self.is_valid():
+
+            seen = {}
+            bucket_array = self.context.object(
+                self.symbol_table_name + constants.BANG + "array",
```

Comment:
This can be called at the start of this method, rather than defining the `__init__` just to cache it in the class?